tests/driver_feetech/main.c

```diff
@@ -0,0 +1,391 @@
+
+static int parse_uart(char *arg)
+{
+    unsigned uart = (unsigned)atoi(arg);
+    if (uart >= UART_NUMOF) {
+        printf("Error: Invalid UART_DEV device specified (%u).\n", uart);
+        return -1;
+    }
+    else if (UART_DEV(uart) == UART_STDIO_DEV) {
+        printf("Error: The selected UART_DEV(%u) is used for the shell!\n", uart);
+        return -2;
+    }
+    return uart;
+}
+
+static int32_t parse_baud(char *arg)
+{
+    int32_t baud = (int32_t)atoi(arg);
+
+    for (size_t i = 0 ; i < ARRAY_LEN(baudrates) ; i++) {
+        if(baud == baudrates[i]) {
+            return baud;
+        }
+    }
+
+    printf("Error: Invalid baudrate (%s)\n", arg);
+    return -1;
+}
+
+static int parse_dev(char *arg)
+{
+    int dev = (int)atoi(arg);
+    if (dev < 0 || 254 < dev) {
+        printf("Error: Invalid device id (%s)\n", arg);
+        return -1;
+    }
+    return dev;
+}
+
+static void parse_reg(char *arg, int *reg8, int *reg16)
+{
+    *reg8 = -1;
+    *reg16 = -1;
+
+    for (size_t i = 0 ; i < ARRAY_LEN(regs8) ; i++) {
+        if(strcmp(arg, regs8[i].name) == 0) {
+            *reg8 = regs8[i].addr;
+            return;
+        }
+    }
+
+    for (size_t i = 0 ; i < ARRAY_LEN(regs16) ; i++) {
+        if(strcmp(arg, regs16[i].name) == 0) {
+            *reg16 = regs16[i].addr;
+            return;
+        }
+    }
+
+    printf("Error: Invalid register (%s)\n", arg);
+}
+
+void print_registers(void) {
+    puts("available 8bits registers :");
+    for (size_t i = 0 ; i < ARRAY_LEN(regs8) ; i++) {
+        printf("\t%s\n", regs8[i].name);
+    }
+
+    puts("available 16bits registers :");
+    for (size_t i = 0 ; i < ARRAY_LEN(regs16) ; i++) {
+        printf("\t%s\n", regs16[i].name);
+    }
+}
+
+static int cmd_init(int argc, char **argv) {
+    int uart = -1;
+    int baud = -1;
+    uint32_t timeout = -1;
+
+    if (argc != 3 && argc != 4) {
+        printf("usage; %s <uart> <baudrate> [<timeout_us>]\n", argv[0]);
+        puts("available baudrates :");
+        for (size_t i = 0 ; i < ARRAY_LEN(baudrates) ; i++) {
+            printf("\t%ld\n", (long int)baudrates[i]);
+        }
+        return 1;
+    }
+    /* parse parameters */
+    uart = parse_uart(argv[1]);
+    if (uart < 0) {
+        return -1;
+    }
+
+    baud = parse_baud(argv[2]);
+    if (baud < 0) {
+        return -1;
+    }
+
+    if (argc == 4) {
+        timeout = (uint32_t)atol(argv[3]);
+        if (timeout == 0) {
+            printf("Error : Invalid timeout (%s)", argv[3]);
+            return -1;
+        }
+    }
+
+    /* init */
+    uart_half_duplex_params_t params = {
+        .uart = uart,
+        .baudrate = baud,
+        .dir = UART_HALF_DUPLEX_DIR_NONE,
+    };
+
+    int ret = uart_half_duplex_init(&stream, feetech_buffer, ARRAY_LEN(feetech_buffer), &params);
+
+    if (argc == 4) {
+        stream.timeout_us = timeout;
+    }
+
+    if (ret == UART_HALF_DUPLEX_NODEV) {
+        puts("Error: invalid UART device given");
```

Comment:
You should return after an error, otherwise you'll also display a success message.